inc/commonitilobject.class.php

```diff
@@ -2594,7 +2593,7 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
                         $linkclass->update($data);
                      }
                   }
-                  $linkclass = new $this->supplierlinkclass();
+                  $linkclass = new $item->supplierlinkclass();
```

Comment:
This part is very strange (your change seems legit btw, $this in static ...)
Changing to $item do a duplicate block, see above lines (2589->2595)

I think we could remove one of these 2 parts